packages/components/form/src/form.ts

```diff
@@ -177,7 +177,7 @@ export class Form<T extends Record<string, any> = Record<string, any>> extends L
   }
 
   #onFormControl(event: SlFormControlEvent): void {
-    const control = event.target;
+    const control = event.composedPath()[0] as EventTarget & HTMLElement & FormControl;
```

Comment:
The type casting without any checks feels a little dangerous. Anyway you could improve this using `instanceof` perhaps?